### DISCUSSION TITLE
Seeking Guidance on Integrating Machine Learning with MOOSE Framework for Nuclear Reactor Optimization Dissertation

### DISCUSSION THREAD

**Discussion by HBurgess173** [idaholab/moose#25172] [Q&A General]
Good afternoon,

I am reaching out to this community in search of help and guidance for my upcoming dissertation. With a background in nuclear mechanical engineering and having been on a degree apprenticeship since 2019, I am now embarking on an exciting research project.

**Background:**
Throughout my academic journey, I've acquired proficiency in modules like CFD, FEA, Engineering Mathematics, MATLAB, Thermofluidic Systems, Stress, and Dynamics. My focus has also extended to nuclear-centric subjects such as reactor physics, reactor chemistry, and reactor materials.

**Project Interest:**
While researching for my project, I came across the NEAMS framework and am intrigued by the possibility of integrating it into my work. My goal is to develop a dissertation topic centered on the use of Machine Learning (ML) or genetic algorithms to optimize a component used in a nuclear reactor.

**Preliminary Idea:**
My initial discussion with my university supervisor has led me to consider the question: "Can ML be used to optimize the design of (BLANK) to increase/decrease the (BLANK)?" An example direction might be "Utilization of Machine Learning for Heat Exchanger Optimization."

**Primary Question:**
My primary question for this community is whether ML can be implemented with the MOOSE framework. If so, who can I contact to discuss how this would work? I am eager to explore this avenue and would greatly appreciate any insights or contacts.

**Request for Feedback:**
I would welcome any comments, suggestions, or discussions about my dissertation proposal, especially if I can align it to current research being carried out in the field. Your expertise and perspectives would be invaluable to my project.

I look forward to your response and thank you in advance for your time and consideration.

Best regards,
Harry Burgess
HBurgess173.HB@gmail.com

**Comment by GiudGiud** [comment]
Hello

We have ML capabilities through our integration of libtorch in the framework. We can build, train and use neural networks inside multiphysics solves essentially. You could leverage this for optimization by relying on a reduced order model (ROM) to perform the optimization steps (and the higher fidelity multiphysics to build that ROM)

I'm assuming you are pursuing doctoral studies in the UK. 
I would recommend you work with your advisor in the UK to devise an optimization problem relevant to the nuclear industry. There are a number of innovative European advanced reactor designs that would be good study topics for example. We use tools based on MOOSE to study US designs already

Best
Guillaume

**Reply by HBurgess173** [reply]
Hi Guillaume,

Thank you for getting back to me.

Currently, I'm working towards my Bachelor's degree, but I have ambitions to delve deeper into this research at the Master's level, which is a step before a doctoral.

Regarding the libtorch framework, do you have any training resources or perhaps a research paper where it's been utilized? I've only recently discovered its potential and am eager to explore its capabilities for my research.

Since I'm relatively new to this domain, could you guide me on how to acquaint myself with cutting-edge European advanced reactor designs and emerging technologies that might be pertinent to my research?

Appreciate your assistance,
Harry

**Reply by GiudGiud** [reply]
@grmnptr for the paper
I think there's a European nuclear society you could engage with. Maybe there is a student chapter or a city chapter near you
https://www.euronuclear.org/
